### DISCUSSION TITLE
Order-fairness consensus protocol (22Q2 brainstorming)

### DISCUSSION THREAD

**Discussion by jsoares** [consensus-shipyard/consensuslab#73] [Project proposals]
- Order-fairness gadget for root consensus (related to Y4)
- Applications of timelock encryption
- Order-Fairness for MEV resilience (subnet)

@sa8 @jsoares @LefKok

**Comment by adlrocha** [comment]
@LefKok, could Narwhal and Tusk help with this problem? I need to read the paper again, but the more I think about it the more I like the idea of having a DAG-based mempool (or having some to have some kind of light consensus stage in the mempool).

**Comment by LefKok** [comment]
This is in direct conflict with deduplication. If we allow full duplication we can find the median time per transaction (if transactions are proposed by >2f+1 parties.
